dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -760,36 +761,50 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
-            if (currentUser != null) {
+        // Try to update GitHub metadata using user's token and getMyself().
+        List<User> gitHubUsers = userDAO.findAllGitHubUsers();
+        LOG.info("Beginning update for " + gitHubUsers.size() + " GitHub users.");
+        for (User u: gitHubUsers) {
+            List<Token> tokens = tokenDAO.findGithubByUserId(u.getId());
+            if (!tokens.isEmpty()) {
                 try {
-                    GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createSourceCodeRepo(t);
-                    gitHubSourceCodeRepo.syncUserMetadataFromGitHub(currentUser, Optional.of(tokenDAO));
-                    LOG.info("Updated " + currentUser.getUsername());
+                    GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createSourceCodeRepo(tokens.get(0));
+                    gitHubSourceCodeRepo.syncUserMetadataFromGitHub(u, Optional.of(tokenDAO));
+                    LOG.info("Updated " + u.getUsername());
                 } catch (Exception ex) {
-                    gitHubUsersNotUpdatedWithToken.add(currentUser);
+                    LOG.info("Could not retrieve GitHub ID for user: " + u.getUsername());
+                    gitHubUsersNotUpdatedWithToken.add(u);
                 }
             }
         }
 
-        // Get the GitHub token of the admin making this call to avoid rate limiting
+        // For users we could not get their GitHub id using their token, get the GitHub id from the avatarurl we have stored.
         LOG.info("Beginning update for " + gitHubUsersNotUpdatedWithToken.size() + " GitHub users who could not be updated with token.");
-        Token t = tokenDAO.findGithubByUserId(user.getId()).get(0);
-        GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createSourceCodeRepo(t);
         for (User u : gitHubUsersNotUpdatedWithToken) {
-            try {
-                gitHubSourceCodeRepo.syncUserMetadataFromGitHubByUsername(u, tokenDAO);
-                LOG.info("Updated " + u.getUsername());
-            } catch (Exception ex) {
-                usersCouldNotBeUpdated.add(u);
-                LOG.info(ex.getMessage(), ex);
+            if (!u.isBanned()) {
+                User.Profile userProfile = u.getUserProfiles().get(TokenType.GITHUB_COM.toString());
+                String avatarUrl = userProfile.avatarURL;
+
+                if (avatarUrl != null) {
+                    Matcher m = GITHUB_ID_PATTERN.matcher(avatarUrl);
+                    if (m.matches()) {
+                        String id = m.group(1);
+                        u.getUserProfiles();
+                        userProfile.onlineProfileId = id;
+                        List<Token> userTokens = tokenDAO.findGithubByUserId(u.getId());
+                        if (!userTokens.isEmpty()) {
+                            userTokens.get(0).setOnlineProfileId(id);
```

Comment:
I guess there's never more than one token, but if there were, you would need to set it for all of them. I guess we do have this pattern of assuming there's at most one throughout the code. But you could just use `forEach`